src/main/java/com/redhat/devtools/lsp4ij/LSPIJEditorUtils.java

```diff
@@ -147,27 +146,27 @@ private static Set<Character> getAbstractFileTypeQuoteCharacters(@NotNull PsiFil
 
     @Nullable
     private static Set<Character> getTextMateQuoteCharacters(@NotNull PsiFile file) {
-        /* TODO: See TextMate versioning comment above
-        Editor editor = isTextMateFile(file) ? LSPIJUtils.editorForElement(file) : null;
-        TextMateScope scope = editor instanceof EditorEx editorEx ? TextMateEditorUtils.getCurrentScopeSelector(editorEx) : null;
-        List<Preferences> allPreferences = scope != null ? TextMateService.getInstance().getPreferenceRegistry().getPreferences(scope) : Collections.emptyList();
-        Preferences preferences = ContainerUtil.getFirstItem(allPreferences);
-        Set<TextMateAutoClosingPair> smartTypingPairs = preferences != null ? preferences.getSmartTypingPairs() : Collections.emptySet();
-        if (!ContainerUtil.isEmpty(smartTypingPairs)) {
-            Set<Character> quoteCharacters = new LinkedHashSet<>();
-            for (TextMateAutoClosingPair smartTypingPair : smartTypingPairs) {
-                CharSequence left = smartTypingPair.getLeft();
-                CharSequence right = smartTypingPair.getRight();
-                if ((left.length() == 1) && left.equals(right)) {
-                    char quoteCharacterCandidate = left.charAt(0);
-                    if (DEFAULT_QUOTE_CHARACTERS.contains(quoteCharacterCandidate)) {
-                        quoteCharacters.add(quoteCharacterCandidate);
+        if (isTextMateFile(file)) {
+            Editor editor = LSPIJUtils.editorForElement(file);
+            TextMateScope scope = editor instanceof EditorEx editorEx ? TextMateEditorUtils.getCurrentScopeSelector(editorEx) : null;
+            List<Preferences> allPreferences = scope != null ? TextMateService.getInstance().getPreferenceRegistry().getPreferences(scope) : Collections.emptyList();
+            Preferences preferences = ContainerUtil.getFirstItem(allPreferences);
+            Set<TextMateAutoClosingPair> smartTypingPairs = preferences != null ? preferences.getSmartTypingPairs() : Collections.emptySet();
```

Comment:
The supported quote characters are in the TextMate bundle's smart typing pairs, but not explicitly called out as being quote characters. As a result, we basically have to find the intersection of the values from the TextMate smart typing pairs and the hard-coded default quote characters.

This means that if some TextMate-supported language uses something other than single quote, double quote, or backtick, it _would be missed_ here, but I'm guessing that's VERY unlikely, at least for single-character-bounded string literals. Obviously something like a text block that's bounded by `"""` wouldn't be included here, but those require a more complex parser than we have for our `inStringLiteral()` utility method anyway.